### PR TITLE
Move core ign config into NodeNool controller

### PR DESCRIPTION
**Pull request by enxebre**
This drops the pod bash hack in favour of putting the logic to fetch the core ignition config in the NodePool controller.
This makes the process and validation consistent fo any config, custom and core.

### CODE REVIEW

**Comment by enxebre** [comment]
I just realiased we need to block ignition token generation and so NodePool creation on this core configs being available. Otherwise there's a race and the ign payload could miss them.

**Comment by ironcladlou** [comment]
Thanks!

/lgtm

File: hypershift-operator/controllers/nodepool/nodepool_controller.go [hunk 1]
```diff
@@ -378,7 +379,10 @@ func (r *NodePoolReconciler) reconcile(ctx context.Context, hcluster *hyperv1.Ho
 	})
 
 	// Validate config input.
-	config, err := r.getConfig(ctx, nodePool)
+	// 3 expectedCoreConfigResources: fips, ssh and haproxy.
+	// TODO (alberto): consider moving the expectedCoreConfigResources check
+	// into the token Secret controller so we don't block Machine infra creation on this.
+	config, err := r.getConfig(ctx, nodePool, 3)
```
Comment (relyt0925) [inline hypershift-operator/controllers/nodepool/nodepool_controller.go:385]: ibm will have more core resources

Comment (relyt0925) [reply hypershift-operator/controllers/nodepool/nodepool_controller.go:385]: (as long as check doesn't fail if there are more than 3 this will work).

**Comment by relyt0925** [comment]
Can we in this PR (or a follow up PR) get a reconciliation path that can react to changes of the core configmaps?

File: hypershift-operator/controllers/nodepool/nodepool_controller.go (hunk 1 above)
Comment (relyt0925) [reply hypershift-operator/controllers/nodepool/nodepool_controller.go:385]: It be nice if that could be "configurable" at least on a per provider basis. (aka IBM Cloud, AWS, etc)

**Comment by relyt0925** [comment]
ignore my comments I'm good with this

**Pull request merged**